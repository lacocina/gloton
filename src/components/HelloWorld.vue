<template>
  <h1>{{ msg }}</h1>

  <div class="card m-5 text-2xl bg-green-500">
    <button type="button" @click="count++">count is {{ count }}</button>
    <p>
      Edit
      <code>components/HelloWorld.vue</code> to test HMR
    </p>
  </div>

  <p class="icon is-twitter">
    Check out
    <a href="https://vuejs.org/guide/quick-start.html#local" target="_blank"
      >create-vue</a
    >, the official Vue + Vite starter
  </p>

  <label>
    <input type="checkbox" class="accent-pink-500" checked> Customized
  </label>

  <select>
    <option>Yes</option>
    <option>No</option>
    <option>Maybe</option>
  </select>

  <select class="appearance-none">
    <option>Yes</option>
    <option>No</option>
    <option>Maybe</option>
  </select>

  <p class="bg-primary">primary</p>
  <p class="hover:bg-primary">hover:primary</p>
  <p class="bg-blue">blue</p>
  <p class="hover:bg-blue transition-colors duration-700">hover:blue</p>
  <p class="bg-blue-50">blue 50</p>
  <p class="bg-blue-100">blue100</p>
  <p class="bg-blue-200">blue200</p>
  <p class="bg-blue-300">blue300</p>
  <p class="bg-blue-400">blue400</p>
  <p class="bg-blue-500">blue500</p>
  <p class="bg-blue-600">blue600</p>
  <p class="bg-blue-700">blue700</p>
  <p class="bg-blue-800">blue800</p>
  <p class="bg-blue-900">blue900</p>
  <p class="read-the-docs">Click on the Vite and Vue logos to learn more</p>
  <p class="read-the-docs read-the-docs--brand">Click on the Vite and Vue logos to learn more</p>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'HelloWorld',
  props: {
    msg: {
      type: String,
      required: true,
    },
  },
  data() {
    return {
      count: 0,
    }
  },
})
</script>

<style scoped>

.read-the-docs {
  color: red;
}
</style>
