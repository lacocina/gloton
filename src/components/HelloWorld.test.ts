import { mount } from '@vue/test-utils'
import HelloWorld from '../components/HelloWorld.vue'

test('mount component', async () => {
  expect(HelloWorld).toBeTruthy()

  const wrapper = mount(HelloWorld, {
    props: {
      msg: "Hello world",
    },
  })

  expect(wrapper.text()).toContain('Hello world')
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('count is 1')

  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('count is 2')
})