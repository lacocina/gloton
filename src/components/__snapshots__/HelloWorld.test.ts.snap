// Vitest Snapshot v1

exports[`mount component 1`] = `
"<h1 data-v-e17ea971=\\"\\">Hello world</h1>
<div class=\\"card\\" data-v-e17ea971=\\"\\"><button type=\\"button\\" data-v-e17ea971=\\"\\">count is 0</button>
  <p data-v-e17ea971=\\"\\"> Edit <code data-v-e17ea971=\\"\\">components/HelloWorld.vue</code> to test HMR </p>
</div>
<p data-v-e17ea971=\\"\\"> Check out <a href=\\"https://vuejs.org/guide/quick-start.html#local\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">create-vue</a>, the official Vue + Vite starter </p>
<p data-v-e17ea971=\\"\\"> Install <a href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Volar</a> in your IDE for a better DX </p>
<p class=\\"read-the-docs\\" data-v-e17ea971=\\"\\">Click on the Vite and Vue logos to learn more</p>"
`;
